src/python_testing/TC_DeviceBasicComposition.py

```diff
@@ -560,54 +666,73 @@ class RequiredMandatoryAttribute:
+                        success = False
 
-                        attribute_string = self.cluster_mapper.get_attribute_string(cluster_id, attribute_id)
-                        logging.info(
-                            f"Checking presence of claimed supported {attribute_string} on {location.as_cluster_string(self.cluster_mapper)}: {'found' if has_attribute else 'not_found'}")
+        self.print_step(8, "Validate that all cluster ID prefixes are in the standard, MEI or test vendor range")
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            cluster_prefixes = [a & 0xFFFF_0000 for a in endpoint.keys()]
+            bad_clusters_ids = [a for a in cluster_prefixes if a >= bad_prefix_min]
+            for bad in bad_clusters_ids:
+                location = ClusterPathLocation(endpoint_id=endpoint_id, cluster_id=bad)
+                self.record_error(self.get_test_name(), location=location,
+                                  problem=f'Bad cluster id prefix {bad}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                success = False
 
-                        # Check attribute is actually present.
-                        if not has_attribute:
-                            # TODO: Handle detecting write-only attributes from schema.
-                            if "WriteOnly" in attribute_string:
-                                continue
+        self.print_step(9, "Validate that all clusters in the standard range have a known cluster ID")
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            standard_clusters = [a for a in endpoint.keys() if a < mei_range_min]
+            unknown_clusters = sorted(list(set(standard_clusters) - set(chip.clusters.ClusterObjects.ALL_CLUSTERS)))
+            for bad in unknown_clusters:
+                location = ClusterPathLocation(endpoint_id=endpoint_id, cluster_id=bad)
+                self.record_error(self.get_test_name(
+                ), location=location, problem=f'Unknown cluster ID in the standard range {bad}', spec_location='Manufacturer Extensible Identifier (MEI)')
+                success = False
 
-                            self.record_error(self.get_test_name(), location=location,
-                                              problem=f"Did not find {attribute_string} on {location.as_cluster_string(self.cluster_mapper)} when it was claimed in AttributeList ({attribute_list})", spec_location="AttributeList Attribute")
-                            success = False
-                            continue
+        self.print_step(10, "Validate that all clusters in the MEI range have a suffix in the manufacturer suffix range")
+        for endpoint_id, endpoint in self.endpoints_tlv.items():
+            mei_clusters = [a for a in endpoint.keys() if a >= mei_range_min]
```

Comment:
does the mei list of thing must be provided by the manufacturer in order to validate this test ?

Reply:
no - manufacturers are free to use any ID in this range and they're responsible for validating that they're backwards compatible. The reason we don't want to allow anything weird in the standard range is because those devices would not be backwards compatible to other controllers if the spec extended the range out. If a manufacturer did something weird, it's local to their own devices.